lametro/utils.py

```diff
@@ -28,37 +29,50 @@ def parse_subject(text):
-    earliest_start = found_events.first().start_time
-    latest_start = found_events.last().start_time
-    # Sometimes multiple events happen at the same time (though in reality, they occur one-after-the-other). Example: the committee events at 1:00 on 05/17/2017.
-    # Check for this situation, and then determine if these events include a board meeting.
-    # If current events do not include a board meeting, the end time should be one hour.
-    if earliest_start == latest_start:
-        if found_events.filter(name__icontains='Board Meeting'):
-            # Custom order: show the board meeting first.
-            # '.annotate' adds a field called 'val', which contains a boolean – we order in reverse, since
-            # false comes before true.
-            return found_events.annotate(val=RawSQL("name like %s", ('%Board Meeting%',))).order_by('-val')
-        else:
-            one_hour_ago = now_with_buffer - timedelta(hours=1)
-            
-            return found_events.filter(start_time__gt=one_hour_ago)
-
-    else:
-        # To find committee events...
-        event_names = [e.name for e in found_events if ("Committee" in e.name) or ("LA SAFE" in e.name) or ("Budget Public Hearing" in e.name) or ("Fare Subsidy Program Public Hearing" in e.name) or ("Crenshaw Project Corporation" in e.name)]
-
-        if event_names:
-            # Set meeting time to one hour
-            one_hour_ago = now_with_buffer - timedelta(hours=1)
-            
-            found_events = found_events.filter(start_time__gt=one_hour_ago)
-
-            if len(found_events) > 1:
-                return found_events
-            else:
-                return found_events.first()
-        else:
-            return found_events.first()                
\ No newline at end of file
+    # Metro provided a spreadsheet of average meeting times. The minimum average time a meeting lasts is 52 minutes: let's round to 55 and remove the 5-minute buffer. 
+    time_ago = now_with_buffer - timedelta(minutes=60)
+    # Custom order: show the board meeting first, when there is one.
+    # '.annotate' adds a field called 'val', which contains a boolean – we order in reverse, since false comes before true.
+    found_events = found_events.annotate(val=RawSQL("name like %s", ('%Board Meeting%',))).order_by('-val')
+
+    if found_events.filter(start_time__gte=time_ago):
+        # Check if previous event is still going on in Legistar.
+        previous_meeting = found_events.filter(start_time__gte=time_ago).last().get_previous_by_start_time()
+        if legistar_meeting_progress(previous_meeting):
+            return Event.objects.filter(ocd_id=previous_meeting.ocd_id)
+
+        return found_events.filter(start_time__gte=time_ago)  
+    else:  
+        # The IF statement handles the below cases:
+        # (1) found_events includes just one event object. Example: the first committee meeting of the day - 9:00 am on 01/18/2018
```

Comment:
Can we just check whether `len(found_events) == 1` and return the event or None (if it's not "In progress"), prior to doing any of the other checks?

Reply:
Well, if it's not 'In progress,' then we need to check if a previous event is. That means potentially pinging Legistar twice on one page load. I am trying to avoid that.